app/models/document.rb

```diff
@@ -1,13 +1,9 @@
 class Document < ApplicationRecord
-  has_attached_file :attachment,
-                    path: ':rails_root/public/system/:attachment/:id/:style/:filename',
-                    url: '/system/:attachment/:id/:style/:filename'
-
+  has_one_attached :attachment
+  validates :attachment, attached: true,
+                         content_type: ['application/pdf', 'application/vnd.ms-excel', 'application/msword', 'application/vnd.openxmlformats-officedocument.wordprocessingml.document', 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet']
```

Comment:
I know you just took it over, but using [mime-types-data](https://rubygems.org/gems/mime-types-data) (the project has the gem already installed) would perhaps be better.

It doesn't give its wisdom that straight easily, but actually the info is much better:

```ruby
%w[xls xlsx doc docx ods odt pdf].flat_map do |ext| 
  MIME::Types.select { |type| type.extensions.include?(ext) }
end.map(&:to_s)
```

And this gives you:

```ruby
[
  "application/excel",
  "application/vnd.ms-excel",
  "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
  "application/msword",
  "application/word",
  "application/x-msword",
  "application/x-word",
  "text/plain",
  "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
  "application/vnd.oasis.opendocument.spreadsheet",
  "application/vnd.oasis.opendocument.text",
  "application/pdf"
]
```

Reply:
but dont confuse `MIME` and `Mime`, since they're not the same!

Reply:
I would make a helper class method in `ApplicationRecord` or so.

Reply:
thank you, good idea